planner/optimize.go

```diff
@@ -271,7 +271,7 @@ func (e *paramMarkerChecker) Leave(in ast.Node) (ast.Node, bool) {
 //  1. ctx is auto commit tagged.
 //  2. plan is point get by pk.
 func isPointGetWithoutDoubleRead(ctx sessionctx.Context, p plannercore.Plan) bool {
-	if !ctx.GetSessionVars().IsAutocommit() {
+	if !plannercore.IsAutoCommitTxn(ctx) {
```

Comment:
this is a bug fix? +1

Reply:
If the session is already `InTxn`, we should avoid prepare the txn once again. The current code is OK for master, but it will create the future once again in this PR.